### PR TITLE
Add `RouteResponse` and tests for it.

### PR DESCRIPTION
**Pull request by siarheishut**

### CODE REVIEW

File: src/request_processor.h [hunk 1]
```diff
@@ -10,6 +10,8 @@
 namespace rm {
 json::Dict ToJson(std::optional<BusResponse> resp, int id);
 json::Dict ToJson(std::optional<StopResponse> resp, int id);
+json::List ToJson(std::vector<RouteResponse::Item> response_items);
```
Comment (foxfrikses) [inline src/request_processor.h:13]: I don't think you need to add `RouteResponse::Item` here, as you're not gonna use it separately from the function below nor test it.

File: src/request_processor.h [hunk 2]
```diff
@@ -10,6 +10,8 @@
...
+json::Dict ToJson(std::optional<RouteInfo> resp, int id);
```
Comment (foxfrikses) [inline src/request_processor.h:14]: `RouteResponse`

File: src/request_processor.h (hunk 1 above)
Comment (siarheishut) [reply src/request_processor.h:13]: Done.

File: src/request_processor.h (hunk 2 above)
Comment (siarheishut) [reply src/request_processor.h]: Done.

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**